woocommerce/woocommerce
# Render order edit page

### All Submissions:






### Changes proposed in this Pull Request:

Implements an order edit screen which is based on the WP's default post edit screen. With COT, we can't use the default post edit screen since it needs a post object to work properly. This PR implements a lightweight version of `edit-form-advanced.php` with reusing code as much as possible.

Existing order meta boxes are also modified to work with both orders and post objects.

Note that this only implements rendering, saving an order will not work just with this PR.



Closes #32471  .

### How to test the changes in this Pull Request:

1. Use the instructions in #32864 to enable the COT-based UI.
2. Click on any order to open the order edit page.
3. Check that data is populating correctly in all default meta boxes.

Note that since this uses a lot of current code with some changes, it's also necessary to test out the current order edit screen. To do that:
1. Switch back to post-based data store from Settings > Advanced > DataStores.
2. Refresh the WooCommerce homepage so that correct links for the order list and edit tables are populated.
3. Go to the edit screen for any order. Check that meta box functionality, especially for file download and custom meta box is working as expected.

### Known issue
- Currently, refund data will not populate on the refund screen since the functionality is not yet implemented. 

### Other information:




### FOR PR REVIEWER ONLY:
